src/dbt_client/dbtCoreIntegration.ts

```diff
@@ -0,0 +1,521 @@
+      this.telemetry.sendTelemetryError(
+        "pythonBridgeCannotParseProjectUnknownError",
+        exc,
+        {
+          adapter: this.adapterType,
+        },
+      );
+      window.showErrorMessage(
+        extendErrorWithSupportLinks(
+          "An error occured while rebuilding the dbt manifest: " + exc + ".",
+        ),
+      );
+    }
+  }
+
+  async findVersion(): Promise<number[]> {
+    return this.python?.lock<number[]>(
+      (python) => python!`to_dict(project.get_dbt_version())`,
+    );
+  }
+
+  async runModel(command: DBTCommand) {
+    command.addArgument("--profiles-dir");
+    command.addArgument(this.dbtProfilesDir);
+    command.addArgument("--project-dir");
+    command.addArgument(this.projectRoot.fsPath);
+    this.executionInfrastructure.addCommandToQueue(command);
+  }
+
+  async buildModel(command: DBTCommand) {
+    command.addArgument("--profiles-dir");
+    command.addArgument(this.dbtProfilesDir);
+    command.addArgument("--project-dir");
+    command.addArgument(this.projectRoot.fsPath);
+    this.executionInfrastructure.addCommandToQueue(command);
+  }
+
+  async runTest(command: DBTCommand) {
+    command.addArgument("--profiles-dir");
+    command.addArgument(this.dbtProfilesDir);
+    command.addArgument("--project-dir");
+    command.addArgument(this.projectRoot.fsPath);
+    this.executionInfrastructure.addCommandToQueue(command);
+  }
+
+  async runModelTest(command: DBTCommand) {
+    command.addArgument("--profiles-dir");
+    command.addArgument(this.dbtProfilesDir);
+    command.addArgument("--project-dir");
+    command.addArgument(this.projectRoot.fsPath);
+    this.executionInfrastructure.addCommandToQueue(command);
+  }
+
+  async compileModel(command: DBTCommand) {
+    command.addArgument("--profiles-dir");
+    command.addArgument(this.dbtProfilesDir);
+    command.addArgument("--project-dir");
+    command.addArgument(this.projectRoot.fsPath);
+    this.executionInfrastructure.addCommandToQueue(command);
+  }
+
+  async generateDocs(command: DBTCommand) {
+    command.addArgument("--profiles-dir");
+    command.addArgument(this.dbtProfilesDir);
+    command.addArgument("--project-dir");
+    command.addArgument(this.projectRoot.fsPath);
+    this.executionInfrastructure.addCommandToQueue(command);
+  }
+
+  async deps(command: DBTCommand) {
+    command.addArgument("--profiles-dir");
+    command.addArgument(this.dbtProfilesDir);
+    command.addArgument("--project-dir");
+    command.addArgument(this.projectRoot.fsPath);
+    this.executionInfrastructure.addCommandToQueue(command);
+  }
+
+  async debug(command: DBTCommand) {
+    command.addArgument("--profiles-dir");
+    command.addArgument(this.dbtProfilesDir);
+    command.addArgument("--project-dir");
+    command.addArgument(this.projectRoot.fsPath);
+    this.executionInfrastructure.addCommandToQueue(command);
+  }
+
+  // internal commands
+  async unsafeCompileNode(modelName: string): Promise<string | undefined> {
```

Comment:
correct me if i am wrong: these internal commands will be common for any integration right? whether it is dbt core or cloud? should nt we move this to core infrastructure?

Reply:
The are not common as they will require different implementation.